tt_metal/impl/buffers/buffer.cpp

```diff
@@ -289,7 +315,7 @@ bool Buffer::is_allocated() const {
 }
 
 uint32_t Buffer::address() const {
-    if (device_->can_use_passthrough_scheduling()) {
+    if (allocation_status_.load(std::memory_order::acquire) != AllocationStatus::ALLOCATION_REQUESTED) {
```

Comment:
I'd say person asking address is always expecting everything is fine and I'd put it under assert and thats all.